### DISCUSSION TITLE
How can i check is there specific channel in discord server?

### DISCUSSION THREAD

**Discussion by 5pread** [Rapptz/discord.py#5841] [Q&A] [answered]
Im trying to make a registery bot, for usage firstly bot have to check is there already have registery channel in discord server. 
I tried to make it with for loops first but they didnt work well actually

```
@client.command()
        @commands.has_permissions(administrator=True, manage_messages=True)
        async def registery(ctx, channelname):
            text_channel_list = [ ]
            for guild in client.guilds:
                for channel in guild.text_channels:
                    text_channel_list.append(channel)
                    if channel.name == channelname:
                        await ctx.send('The server have that channel')
                    else:
                        await ctx.send('F')

```
Then i tried to get '**registery**' channel with discord.Utils.get command but seems like it wasnt work like the other command.

```
def checkforchannel(channelname_to_check):
            channel = discord.utils.get(client.get_all_channels(), name=channelname_to_check)
            if channel == channelname_to_check:
                return True
            else:
                return False
```
In documentation it says if bot cant able to find/get channel return output is None, but i always get False return from this code whenever i run it, so probably it returns None everytime. 

Any suggestions?

**Comment by Rapptz** [comment, marked as answer]
[`discord.utils.get`](https://discordpy.readthedocs.io/en/latest/api.html#discord.utils.get) gets the object or `None` if it's not found. So `channel` is either a `discord.abc.GuildChannel` (which can either be `discord.TextChannel`, `discord.CategoryChannel`, `discord.StoreChannel`, or `discord.VoiceChannel`) or `None`.

However you're comparing it to a string which will always be `False`. To fix your code do the following:

```py
def checkforchannel(channelname_to_check):
    channel = discord.utils.get(client.get_all_channels(), name=channelname_to_check)
    return channel is not None
```

Note however as I hinted on above that `get_all_channels` goes through *all* channels in every guild your bot is on. It's probably a good idea to restrict the search to something that either belongs to a specific guild using its guild ID:

```py
def check_for_channel(channel_name):
    # Replace with your guild's guild ID
    guild = client.get_guild(81384788765712384)
    if guild is None:
        # This can happen if the lookup failed
        return False

    channel = discord.utils.get(guild.channels, name=channel_name)
    return channel is not None
```

You can further restrict this by using `guild.text_channels` if you only care about `discord.TextChannel` or `guild.voice_channels` if you only care about `discord.VoiceChannel`.

**Reply by 5pread** [reply]
Thanks for answering, its working well rn

**Comment by Mouly22** [comment]
I did not know it! Thanks for sharing @Rapptz
